server/src/main/java/org/elasticsearch/index/fielddata/ordinals/SingletonGlobalOrdinalMapping.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+package org.elasticsearch.index.fielddata.ordinals;
+
+import org.apache.lucene.index.OrdinalMap;
+import org.apache.lucene.index.SortedDocValues;
+import org.apache.lucene.index.TermsEnum;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.LongValues;
+
+import java.io.IOException;
+
+/**
+ * {@link SortedDocValues} implementation that returns global ordinals instead
+ * of segment ordinals.
+ */
+class SingletonGlobalOrdinalMapping extends SortedDocValues {
+    private final SortedDocValues values;
+    private final OrdinalMap ordinalMap;
+    private final LongValues mapping;
+    private final TermsEnum[] lookups;
+
+    SingletonGlobalOrdinalMapping(OrdinalMap ordinalMap, SortedDocValues values, TermsEnum[] lookups, int segmentIndex) {
+        this.values = values;
+        this.lookups = lookups;
+        this.ordinalMap = ordinalMap;
```

Comment:
I agree with Igor.  At a minimum we should throw here if `ordinalMap.getValueCount() >= MAX_INT`.  I think it's okay if we require callers to pre-check it rather than building a factory method (although a factory method would also be fine), but we shouldn't allow the construction of invalid objects.

Reply:
Done!